security/advancedtls/advancedtls_test.go

```diff
@@ -679,3 +681,105 @@ func TestOptionsConfig(t *testing.T) {
 		})
 	}
 }
+
+const (
+	tlsEcdheRsaWith3desEdeCbcSha uint16 = 0xc012
+)
+
+const (
+	CurveP256 tls.CurveID = 23
+)
+
+const (
+	pointFormatUncompressed uint8 = 0
+)
+
+const (
+	VersionTLS10 uint16 = 0x0301
+)
+
+func TestGetCertificateSNI(t *testing.T) {
+	serverPeerCert1, err := tls.LoadX509KeyPair(testdata.Path("server_cert_1.pem"),
+		testdata.Path("server_key_1.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	serverPeerCert2, err := tls.LoadX509KeyPair(testdata.Path("server_cert_2.pem"),
+		testdata.Path("server_key_2.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	serverPeerCert3, err := tls.LoadX509KeyPair(testdata.Path("server_cert_3.pem"),
+		testdata.Path("server_key_3.pem"))
+	if err != nil {
+		t.Fatalf("Server is unable to parse peer certificates. Error: %v", err)
+	}
+	tests := []struct {
+		desc          string
+		serverGetCert func(*tls.ClientHelloInfo) ([]*tls.Certificate, error)
+	}{
+		{
+			desc: "Select the certificate that matches the server name provided in clientHello",
+			serverGetCert: func(info *tls.ClientHelloInfo) ([]*tls.Certificate, error) {
+				return []*tls.Certificate{&serverPeerCert1, &serverPeerCert2, &serverPeerCert3}, nil
+			},
+		},
+	}
+	for _, test := range tests {
+		test := test
```

Comment:
For testing, I would suggest the following: we set three certificates when setting the callback(since now we can set a list of certs), and in the clientHello we specify one of them. Then we use serverConfig.GetCertificate() to check if the one we want is selected. WDYT?

Reply:
I changed the structure of test cases a bit. Could you please check it again and see if that's what you meant? Thank you!